### PR TITLE
Validation_model_specs

### PR DESCRIPTION
**Pull request by AJ193**
### Project requirements
- Add the following validations:
- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb
Comment (fobadara) [inline spec/models/user_spec.rb:1]: [REQUIRED] It is a requirement to test all custom methods in a model for example the `recent_posts` method you created. Kindly fix it and ensure that all your tests are passing

File: spec/models/post_spec.rb
Comment (fobadara) [inline spec/models/post_spec.rb:1]: [REQUIRED] It is a requirement to test all custom methods in a model for example the `recent_comments` and  `update_counter` methods you created. Kindly fix it and ensure that all your tests are passing.

File: spec/models/comment_spec.rb
Comment (fobadara) [inline spec/models/comment_spec.rb:1]: [REQUIRED] It is a requirement to test all custom methods in a model for example the `update_counter` method you created. Kindly fix it and ensure that all your tests are passing.

File: app/models/like.rb
Comment (fobadara) [inline app/models/like.rb:1]: [REQUIRED] Ensure that LikesCounter is an integer greater than or equal to zero

File: spec/models/like_spec.rb
Comment (fobadara) [inline spec/models/like_spec.rb:1]: [REQUIRED] Kindly include tests for the validation of LikesCounter being an integer greater than or equal to zero.

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @AJ193, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt
✅ Good implementation of the principle of separating concern
✅ Use of describe blocks

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AJ193** [comment]
@fobadara Hello thank you for reviewing my code. I want to inform that all my tests are passing correctly now and there is no issue in the tests folder and here is the screenshot of the tests that all of them are passing.
Review 3

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @AJ193 🙋‍♀️,  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
 
You have submitted a project, that does does not have the required changes that are requested by the previous reviewer.

- ***Kindly add all the test that are requested by the previous reviewer. It was requested that you add tests for all public methods in your models as per project requirements.***

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by descholar-ceo — APPROVED** [review]
Hi @AJ193 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 You have implemented all test cases as it has been requested from the previous reviews, and the test cases passed. That's commendable 💪🏽 
👍🏽 Your communication is good, that's really good in real-world industry will help you ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**